torch_geometric/sampler/neighbor_sampler.py

```diff
@@ -365,57 +373,94 @@ def sample_from_edges(
+                    seed_dst, inverse_dst = seed_dst.unique(
+                        return_inverse=True)
+                    edge_label_index = torch.stack([inverse_src, inverse_dst],
+                                                   0)
+                    seed_dict = {
+                        self.input_type[0]: seed_src,
+                        self.input_type[-1]: seed_dst,
+                    }
+                    if edge_label_time is not None:
+                        seed_time_dict = {
+                            self.input_type[0]:
+                            scatter_min(edge_label_time, inverse_src)[0],
+                            self.input_type[-1]:
+                            scatter_min(edge_label_time, inverse_dst)[0],
+                        }
 
             else:  # Merge both source and destination node indices:
-                seed_nodes = edge_label_index.view(-1)
-                seed_nodes, inverse = seed_nodes.unique(return_inverse=True)
-                edge_label_index = inverse.view(2, -1)
-                seed_dict = {self.input_type[0]: seed_nodes}
-                if edge_label_time is not None:
-                    tmp = torch.cat([edge_label_time, edge_label_time])
-                    seed_time_dict = {
-                        self.input_type[0]: scatter_min(tmp, inverse)[0]
-                    }
+                if self.disjoint_sampling:
+                    seed_nodes = edge_label_index.view(-1)
+                    edge_label_index = torch.arange(0,
+                                                    2 * num_seed_edges).view(
+                                                        2, -1)
+                    seed_dict = {self.input_type[0]: seed_nodes}
+                    if edge_label_time is not None:
+                        tmp = torch.cat([edge_label_time, edge_label_time])
+                        seed_time_dict = {self.input_type[0]: tmp}
+                else:
+                    seed_nodes = edge_label_index.view(-1)
+                    seed_nodes, inverse = seed_nodes.unique(
+                        return_inverse=True)
+                    edge_label_index = inverse.view(2, -1)
+                    seed_dict = {self.input_type[0]: seed_nodes}
+                    if edge_label_time is not None:
+                        tmp = torch.cat([edge_label_time, edge_label_time])
+                        seed_time_dict = {
+                            self.input_type[0]: scatter_min(tmp, inverse)[0]
+                        }
 
             output = self._sample(
                 seed=seed_dict,
                 seed_time_dict=seed_time_dict,
             )
 
+            if self.disjoint_sampling:
+                output.batch = update_batch_for_sample_edge(
+                    output.batch, num_seed_edges)
             output.metadata = (edge_label_index, edge_label, edge_label_time)
 
         elif issubclass(self.data_cls, Data):
             assert self.node_time is None  # TODO
```

Comment:
I think we could support this now easily? Let's drop the `assert` and later pass `self._sample(seed=seed_nodes, seed_time=self.node_time)`. WDYT?

Reply:
You mean support batch dict when we don't do disjoint sampling. Do we even want to support that, because the batch dict is meaningless.

Reply:
No, I mean temporal sampling in homogeneous graphs.